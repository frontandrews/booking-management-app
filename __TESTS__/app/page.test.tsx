/* eslint-disable @next/next/no-img-element */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '@/app/page';

jest.mock('next/image', () => {
  const Image = ({ src, alt }: { src: string; alt: string }) => (
    <img src={src} alt={alt} />
  );
  Image.displayName = 'Image';
  return Image;
});
jest.mock('@/components/homepage/main-hero', () => {
  const MainHero = () => <div data-testid="main-hero">MainHero Component</div>;
  MainHero.displayName = 'MainHero';
  return MainHero;
});

jest.mock('@/components/ui/loader-wrap', () => {
  const LoaderWrap = () => (
    <div data-testid="loader-wrap">LoaderWrap Component</div>
  );
  LoaderWrap.displayName = 'LoaderWrap';
  return LoaderWrap;
});

describe('Home Component', () => {
  test('renders Home component correctly', () => {
    render(<Home />);

    expect(screen.getByTestId('main-hero')).toBeInTheDocument();
    expect(screen.getByTestId('loader-wrap')).toBeInTheDocument();

    const titles = screen.getAllByRole('heading', { level: 2 });
    expect(titles.length).toBeGreaterThan(0);

    const descriptions = screen.getAllByText(
      (content, element) => element?.tagName.toLowerCase() === 'p',
    );
    expect(descriptions.length).toBeGreaterThan(0);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
