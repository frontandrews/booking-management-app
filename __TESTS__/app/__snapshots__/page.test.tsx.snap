// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="loader-wrap"
  >
    LoaderWrap Component
  </div>
  <div
    data-testid="main-hero"
  >
    MainHero Component
  </div>
  <div
    class="container mx-auto mt-8 px-4 slide-up"
  >
    <div
      class="flex flex-col mb-8"
    >
      <div
        class="mb-4"
      >
        <h2
          class="text-2xl font-bold"
        >
          App Overview
        </h2>
        <p
          class="text-gray-700"
        >
          The app is a hosting management tool used to organize and manage property reservations.
        </p>
      </div>
      <div
        class="relative w-full h-64 md:h-96 lg:h-[850px]"
      >
        <img
          alt="App Overview"
          src="/app/app-view.gif"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-8"
    >
      <div
        class="mb-4"
      >
        <h2
          class="text-2xl font-bold"
        >
          Sign In
        </h2>
        <p
          class="text-gray-700"
        >
          You can sign in easily using an email and password. If you don't have an account, the app will automatically create one for you.
        </p>
      </div>
      <div
        class="relative w-full h-64 md:h-96 lg:h-[850px]"
      >
        <img
          alt="Sign In"
          src="/app/creating-new-user.gif"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-8"
    >
      <div
        class="mb-4"
      >
        <h2
          class="text-2xl font-bold"
        >
          Properties
        </h2>
        <p
          class="text-gray-700"
        >
          To create a reservation, the user first needs to create a property. After that, the user can edit and delete properties and view their reservations.
        </p>
      </div>
      <div
        class="relative w-full h-64 md:h-96 lg:h-[850px]"
      >
        <img
          alt="Properties"
          src="/app/creating-and-editing-property.gif"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-8"
    >
      <div
        class="mb-4"
      >
        <h2
          class="text-2xl font-bold"
        >
          Reservations
        </h2>
        <p
          class="text-gray-700"
        >
          Once a property is created, the user can create and edit reservations, modify dates, and remove them. However, reservations for a property cannot overlap.
        </p>
      </div>
      <div
        class="relative w-full h-64 md:h-96 lg:h-[850px]"
      >
        <img
          alt="Reservations"
          src="/app/creating-and-editing-reservation.gif"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-8"
    >
      <div
        class="mb-4"
      >
        <h2
          class="text-2xl font-bold"
        >
          Stats
        </h2>
        <p
          class="text-gray-700"
        >
          The user can view stats about properties, monitor the number of reservations for a month, track the revenue generated by a property, and compare it with previous months.
        </p>
      </div>
      <div
        class="relative w-full h-64 md:h-96 lg:h-[850px]"
      >
        <img
          alt="Stats"
          src="/app/stats.gif"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
